src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -295,32 +342,14 @@ def validate(self, node, options=None):
         else:
             # There should be one index per dimension. This is enforced
             # by the array create method so is not tested here.
-
-            # The first index should be a range. This
-            # transformation is currently limited to Ranges which
-            # specify the full extent of the dimension.
-            if not matrix2.is_full_range(0):
-                raise TransformationError(
-                    f"To use matmul2code_trans on matmul, the first index of "
-                    f"the 2nd argument '{matrix2.name}' must be a full range.")
-            # Check that the second dimension is a full range if it is
-            # a range.
-            if (len(matrix2.symbol.shape) > 1 and
-                    isinstance(matrix2.children[1], Range)
-                    and not matrix2.is_full_range(1)):
+            # For matrix2, we can have 1 or 2 full ranges.
+            full_range_order, _, _ = _get_full_range_split(matrix2)
+            n_full_ranges = len(full_range_order)
+            if n_full_ranges not in [1, 2]:
                 raise TransformationError(
-                    f"To use matmul2code_trans on matmul for a matrix-matrix "
-                    f"multiplication, the second index of the 2nd "
-                    f"argument '{matrix2.name}' must be a full range.")
-            if len(matrix2.children) > 2:
-                # The 3rd index and onwards must not be ranges.
-                for (count, index) in enumerate(matrix2.children[2:]):
-                    if isinstance(index, Range):
-                        raise TransformationError(
-                            f"To use matmul2code_trans on matmul, only the "
-                            f"first two indices of the 2nd argument are "
-                            f"permitted to be a Range but found "
-                            f"{type(index).__name__} at index {count+2}.")
+                    f"To use matmul2code_trans on matmul, one or two "
```

Comment:
self.name and matrix2.debug_string() please.

Reply:
Addressed in 10d5abbfb758dbaa45439829f974a99504e27227.